### PR TITLE
Initial basic cache of fonts in a service worker. 

### PR DESCRIPTION
**Pull request by jmzwar**
This PR implements a basic service worker for loading re-used assets. A list of fonts and images are cached. Also includes removing woff files as ie support no longer required.

### CODE REVIEW

File: pages/_app.tsx [hunk 1]
```diff
@@ -35,6 +35,20 @@ const queryClient = new QueryClient({
...
+					function (registration) {
+						console.log('Service Worker registration successful with scope: ', registration.scope);
+					},
+					function (err) {
+						console.log('Service Worker registration failed: ', err);
+					}
+				);
+			});
+		}
+	}, []);
```
Comment (bachstatter) [inline pages/_app.tsx:51]: Maybe also put it in a hook?

File: public/sw.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		'/fonts/Inter-Regular.woff2',
+		'/fonts/Inter-SemiBold.woff2',
+		'/fonts/Inter-Bold.woff2',
+		'/fonts/GT-America-Mono-Bold.woff2',
+		'/fonts/GT-America-Extended-Bold.woff2',
+		'/fonts/GT-America-Condensed-Medium.woff2',
+		'/fonts/GT-America-Condensed-Bold.woff2',
+	];
+
+	const IMAGES_CACHE = [
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply public/sw.js:16]: These are in the public folder, so they are just dumped into the root directory. Possibly there could be a way to set cache headers on them with next but the service worker can cover it.

[the remark this answers is not in the document]
Comment (jmzwar) [reply public/sw.js:16]: We were also serving fonts through webpack as well but they were duplicates. This was actually what was causing the FOUT.

[the remark this answers is not in the document]
Comment (jmzwar) [reply public/sw.js:16]: I'd say that is a good idea, but let's get this merged first as it removes some of the extra crud we don't need.

File: hooks/useServiceWorker.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { useEffect } from 'react';
+
+const swInit = () => {
+	if ('serviceWorker' in navigator) {
+		navigator.serviceWorker.register('/sw.js').then(
```
[the remark this answers is not in the document]
Comment (jmzwar) [reply hooks/useServiceWorker.ts:5]: That is only an issue if you cache the sw itself. For this PR it is just fonts and some images in the public folder.

**Pull request merged**